wandb/util.py

```diff
@@ -607,11 +607,14 @@ def json_friendly(  # noqa: C901
         obj = obj.item()
         if isinstance(obj, float) and math.isnan(obj):
             obj = None
-        elif isinstance(obj, np.generic) and obj.dtype.kind == "f":
+        elif isinstance(obj, np.generic) and (
+            obj.dtype.kind == "f" or obj.dtype == "bfloat16"
```

Comment:
Note that when `obj.dtype == "bfloat16"`, its `kind == "V"` (stands for "void"), so this is why I'm being this explicit here.

Reply:
I think `np.generic` is only for `numpy` scalar... I think we probably want to fix the array case as well?

Reply:
I think it is always "unwrapped" recursively, so an array of bfloat16's is eventually converted to an array of floats. I expanded the yea test a bit (it passes): https://github.com/wandb/client/pull/3528/commits/fa8ad77f9b40ed0aeb14cc72ec5c5f71574ff77b

Reply:
ok.. nice